features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Comments/Comment.jsx

```diff
@@ -0,0 +1,289 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Typography } from '@material-ui/core/';
+import { withStyles } from '@material-ui/core/styles';
+import AccountBox from '@material-ui/icons/AccountBox';
+import Grid from '@material-ui/core/Grid';
+import Alert from '../../../Shared/Alert';
+import ConfirmDialog from '../../../Shared/ConfirmDialog';
+import CommentAdd from './CommentAdd';
+import CommentEdit from './CommentEdit';
+import CommentOptions from './CommentOptions';
+import CommentReply from './CommentReply';
+import API from '../../../../data/api';
+
+const styles = theme => ({
+    link: {
+        color: theme.palette.getContrastText(theme.palette.background.default),
+        cursor: 'pointer',
+    },
+    commentIcon: {
+        color: theme.palette.getContrastText(theme.palette.background.default),
+    },
+    commentText: {
+        color: theme.palette.getContrastText(theme.palette.background.default),
+        marginTop: theme.spacing.unig,
+        width: '100%',
+        whiteSpace: 'pre-wrap',
+        overflowWrap: 'break-word',
+    },
+    root: {
+        marginTop: theme.spacing.unit * 2.5,
+    },
+    contentWrapper: {
+        maxWidth: theme.custom.contentAreaWidth,
+        paddingLeft: theme.spacing.unit * 2,
+        paddingTop: theme.spacing.unig,
+    },
+});
+
+/**
+   * Display a particular comment and details
+   * @class Comment
+   * @extends {React.Component}
+   */
+class Comment extends React.Component {
+    /**
+     * Creates an instance of Comment
+     * @param {*} props properies passed by the parent element
+     * @memberof Comment
+     */
+    constructor(props) {
+        super(props);
+        this.state = {
+            openDialog: false,
+            replyIndex: -1,
+            editIndex: -1,
+            deleteComment: null,
+        };
+        this.keyCount = 0;
+        this.getKey = this.getKey.bind(this);
+        this.handleClickDeleteComment = this.handleClickDeleteComment.bind(this);
+        this.handleShowEdit = this.handleShowEdit.bind(this);
+        this.handleShowReply = this.handleShowReply.bind(this);
+        this.handleClickOpen = this.handleClickOpen.bind(this);
+        this.showAddComment = this.showAddComment.bind(this);
+        this.showEditComment = this.showEditComment.bind(this);
+        this.handleConfirmDialog = this.handleConfirmDialog.bind(this);
+        this.handleClose = this.handleClose.bind(this);
+        this.filterRemainingComments = this.filterRemainingComments.bind(this);
+        this.filterCommentToDelete = this.filterCommentToDelete.bind(this);
```

Comment:
can we use arrow functions for these functions ? then we don't need to add another explicit bind statement in the constructor